pkg/api/unversioned/helpers.go

```diff
@@ -63,6 +64,52 @@ func LabelSelectorAsSelector(ps *LabelSelector) (labels.Selector, error) {
 	return selector, nil
 }
 
+func ParseToLabelSelector(selector string) (*LabelSelector, error) {
+	p := labels.NewParser(labels.NewLexer(selector, 0))
```

Comment:
I am looking at this now and it looks like moving it to pkg/labels creates an import cycle. We will have to move `LabelSelectorAsSelector()` and `SetAsLabelSelector()` as well.